lychee-lib/src/types/cache.rs

```diff
@@ -19,6 +19,28 @@ pub enum CacheStatus {
     Unsupported,
 }
 
+impl<'de> Deserialize<'de> for CacheStatus {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let status = <&str as Deserialize<'de>>::deserialize(deserializer)?;
+        match status {
+            "Excluded" => Ok(CacheStatus::Excluded),
+            "Unsupported" => Ok(CacheStatus::Unsupported),
```

Comment:
I know that you've covered these with the tests, but do those values actually get serialized?

I've discarded my test container since, but looking at my comment, I shared two `.lycheecache` files. One was with my modifications to use serde attributes and write the enum type followed by the content/value.

Then in a [follow-up comment I shared another cache file](https://github.com/lycheeverse/lychee/issues/647#issuecomment-1170773449) which noted that `Excluded` and `Unsupported` were not written for the serde attribute "untagged".

---

Has something changed since then? Or am I mistaken, and `.lycheecache` does get these enums stored properly? (_perhaps the default for serde doesn't encounter the problem?_)

Reply:
I grabbed a baseline `.lycheecache` file by generating it from what is currently on master right now. Based on that it seems like both `Excluded` and `Unsupported` are written as expected to the file even with serde's defaults
```
 ~/lychee (master) cargo run -- --verbose --cache fixtures/TEST_EXAMPLE_DOMAINS.md                                                                                  
🔍 7 Total ✅ 2 OK 🚫 0 Errors 💤 5 Excluded

 ~/lychee (master) bat .lycheecache                                                                                                                                     
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: .lycheecache
       │ Size: 303 B
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ https://github.com/rust-lang/rust,200,1657687413
   2   │ http://example.net/foo/bar,Excluded,1657687413
   3   │ http://example.org/,Excluded,1657687413
   4   │ http://foo.example.org/,Excluded,1657687413
   5   │ https://www.rust-lang.org/,200,1657687413
   6   │ http://example.com/,Excluded,1657687413
   7   │ https://example.com/,Excluded,1657687413
───────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

That being said I will definitely add some test to ensure the new deserialization logic works with the default serialization

Reply:
I improved the CLI test to be a full integration test of the caching behavior: https://github.com/lycheeverse/lychee/pull/685/commits/e55b8239028217334b2e4a7301845022dca2ca9f

That should hopefully validate the serialization and deserialization of `CacheStatus`